CoursesManager.UI/Service/MailService.cs

```diff
@@ -6,92 +6,53 @@
 using CoursesManager.UI.Models;
 using CoursesManager.MVVM.Mail.MailService;
 using CoursesManager.MVVM.Mail;
-using CoursesManager.MVVM.Mail.MailTemplate;
 using ZstdSharp.Unsafe;
 using System.ComponentModel.DataAnnotations;
 
-public class MailService : IMailService
+public class MailService
 {
-    private readonly SmtpConfig _smtpConfig;
-    //Uncomment wanneer Encryption Service klaar is
-    //private readonly EncryptionService _encryptionService;
+    private SmtpConfig? _smtpConfig;
 
-    public MailService()
-    {
-        var mailConnectionString = EnvManager<EnvModel>.Values.MailConnectionString;
-
-        if (string.IsNullOrWhiteSpace(mailConnectionString))
-        {
-            throw new Exception("MailConnectionString is niet ingesteld in .env");
-        }
-
-        _smtpConfig = ParseConnectionString(mailConnectionString);
-    }
-    public async Task<MailResult> SendMail(IMailTemplate mailTemplate)
+    public async Task<MailResult> SendMail(MailMessage mailMessage)
     {
         try
         {
-            var emailMessage = mailTemplate.GenerateMail();
+            var mailConnectionString = EnvManager<EnvModel>.Values.MailConnectionString;
+
+            if (string.IsNullOrWhiteSpace(mailConnectionString))
+            {
+                throw new Exception("MailConnectionString is niet ingesteld in .env");
+            }
+
+            _smtpConfig = ParseConnectionString(mailConnectionString);
+
             using var smtpClient = new SmtpClient(_smtpConfig.Server, _smtpConfig.Port)
             {
-                Credentials = new NetworkCredential(_smtpConfig.User, _smtpConfig.Password), // _smtpConfig.Password aanpassen naar _encryptionService.Decrypt(_smtpConfig.Password) wanneer EncryptionService klaar is
+                Credentials = new NetworkCredential(_smtpConfig.User, _smtpConfig.Password),
                 EnableSsl = _smtpConfig.EnableSsl
             };
```

Comment:
Zou je hier misschien beter een methode van kunnen maken? Iets van GetSmtpConfig? Dan kun je die een SmtpConfig laten returnen en dan kun je ook dat private veld voor smtp config weghalen.

Reply:
Ongeacht waar ik het maak moet ik nogsteeds private SmtpConfig ophalen dus zie zelf nog niet helemaal het voordeel? kun je misschien iets meer uitleggen wat je bedoeld?

Reply:
Ik had SmtpConfig Model gemaakt die moeg ik dan toch ophalen? dan maakt het toch niet uit of ik het anders in de methode doe? of zie ik dat verkeerd